drivers/ethernet/eth_stm32_hal.c

```diff
@@ -59,22 +59,6 @@ static const struct device *eth_stm32_phy_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0,
 #define ETH_STM32_AUTO_NEGOTIATION_ENABLE                                                          \
 	UTIL_NOT(DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, phy_handle), fixed_link))
```

Comment:
might be better to split the v1 and v2 driver completely.

Reply:
This information needs to be extracted from the device tree. Since DT_DRV_COMPAT does not differentiate between v1 and v2, there is no good way to incorporate this part into eth_stm32_hal_v1.c. 

Do you have any suggestions?

Reply:
both can have the same DT_DRV_COMPAT. V1 and V2 will not be used at the same time, that's depended on the soc.

Also please make it so, that the v2 driver will preserve the git history of the original driver, so that git blame works at least for v2. maybe rename it in one commit and then split v1 out into v1 driver file or something like that

Reply:
I used git mv, but still lost the history of the code in eth_stm32_hal.c and eth_stm32_hal_v1.c. 

```
git mv eth_stm32_hal.c eth_stm32_hal_v2.c
git commit -s

cp eth_stm32_hal_v2.c eth_stm32_hal_v1.c
cp eth_stm32_hal_v2.c eth_stm32_hal.c

clean code eth_stm32_hal_v2.c
clean code eth_stm32_hal_v1.c
clean code eth_stm32_hal.c

git add .
git commit -s
```
Are there any better ways to retain all the historical information?

Reply:
The special code v1 consists of only ~300 lines.

Reply:
It's not 300, it's 600.  Here is a new [PR 94142](https://github.com/zephyrproject-rtos/zephyr/pull/94142) that splits v1/v2 in this way. There is no common eth_stm32_hal.c. To be honest, there is quite a lot of repetitive code.